core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/AggregatedWorkspaceService.scala

```diff
@@ -20,13 +21,67 @@ import org.broadinstitute.dsde.rawls.model.{
 
 import java.util.UUID
 import scala.jdk.CollectionConverters._
+import scala.util.Try
 
 /**
   * Responsible for aggregating workspace data together from various sources (i.e, WSM).
   * @param workspaceManagerDAO DAO for talking to WSM
   */
 class AggregatedWorkspaceService(workspaceManagerDAO: WorkspaceManagerDAO) extends LazyLogging {
 
+  /**
+    * Given a list of workspaces, aggregates any available workspace information from workspace manager (WSM).
+    *
+    * Pre-fetches all available workspace data from WSM for the given context.
+    *
+    * If the WSM workspace is a legacy Rawls workspace, it is assumed to be GCP, and the googleProjectId from the
+    * provided workspace will be echo'd back out, and it will not attempt to match the workspace with WSM data
+    *
+    * Exceptions encountered when matching a multi-cloud/WSM workspace to the data from WSM are reported in the errorMessage of the workspace
+    *
+    * @param workspaces The list of source rawls workspaces
+    * @param ctx Rawls request and tracing context.
+    *
+    * @throws WorkspaceAggregationException when an error is encountered pulling data from aggregating upstream systems
+
```

Comment:
Nit: blank line (I'm not sure if that messes up the JavaDoc).

Reply:
They're ignored by formatters.
Apparently you have to jump through some hoops to actually keep them.